### ISSUE TITLE
Add async eth.get_uncle_count

### ISSUE THREAD

**Issue by kclowes** [ethereum/web3.py#2822]
### What was wrong?

Some methods are still unavailable on the `AsyncEth` class.

### How can it be fixed?

Following the pattern already established, add `get_uncle_count` to the AsyncEth class, and add corresponding tests.

**Comment by konichuvak** [comment]
Would be happy to help with these. 

I assume that the tests should be added [here](https://github.com/ethereum/web3.py/blob/58a8b2c47e240ae82fc540190d8a1d6cc650bd25/web3/_utils/module_testing/eth_module.py#L108).
If that is the case, could you please comment on how to run the tests for `AcyncEthModuleTest`?
Since there is no conftest in any of the modules along the path to the test class, pytest does recognize it as a valid test target.

I see that for `EthModuleTest` we use `TestEthereumTesterEthModule`, individual tests for which could be ran with a command like 
```shell
pytest tests/integration/test_ethereum_tester.py -k "test_eth_getBlockByNumber_finalized"
```

**Comment by kclowes** [comment]
Awesome, thank you!

> I assume that the tests should be added [here](https://github.com/ethereum/web3.py/blob/58a8b2c47e240ae82fc540190d8a1d6cc650bd25/web3/_utils/module_testing/eth_module.py#L108).

Yep, tests should be added in that file. 

> If that is the case, could you please comment on how to run the tests for AcyncEthModuleTest?

You should be able to pretty much copy/paste the sync tests and add `async`s and `await`s where needed. To run the tests locally against Geth, you can run `pytest tests/integration/go_ethereum/test_goethereum_http.py` or to run against `eth-tester` you can run `pytest tests/integration/test_ethereum_tester.py`. And like you mentioned, you can pass the `-k` flag to single out test(s). Or, to run all of the tests in the AsyncEth module, you can do: `pytest tests/integration/go_ethereum/test_goethereum_http.py::TestGoEthereumAsyncEthModuleTest`. Let me know if you have any other questions!

Thanks for the help!

**Comment by kclowes** [comment]
Closed via #2850
